base/complex.jl

```diff
@@ -1028,3 +1028,114 @@ function complex(A::AbstractArray{T}) where T
     end
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
+
+"""
+    div(z1, z2, r::RoundingMode=RoundNearest)
+
+The quotient from Euclidean division of two Gaussian integers. 
+Computes z1/z2, rounded to a Gaussian integer according to the rounding mode `r`. 
+In other words, the quantity
+
+    round(z1/z2,r,r)
+
+without any intermediate rounding.
+I.e. quotient with the same rounding mode `r` applied to real and imaginary parts.
+
+# Examples:
+```jldoctest
+julia> div(4 + 4im, 3 + 1im, RoundDown)
+1 + 0im
+
+julia> div(4 + 4im, 3 + 1im, RoundUp)
+2 + 1im
+
+julia> div(5 + 1im, 2 + 1im, RoundNearest)
+2 - 1im
+
+julia> div(5 + 1im, 2 + 1im, RoundNearestTiesAway)
+2 - 1im
+
+julia> div(-5 + 5im, 2 + 2im, RoundNearest)
+0 + 2im
+
+julia> div(-5 + 5im, 2 + 2im, RoundNearestTiesAway)
+0 + 3im
+
+julia> div(-5 + 5im, 2 + 2im, RoundNearestTiesUp)
+0 + 3im
+```
+"""
+function div(a::Complex{T}, b::Complex{V}, r::RoundingMode=RoundNearest) where {T<:Integer, V<:Integer}
```

Comment:
`RoundNearest` shouldn't be the default here, because `div` for real integers always rounds down. I am wondering whether all rounding modes actually make sense for complex numbers though.

Reply:
I kept RoundNearest as the default rounding mode since Euclidean GCD algorithm requires RoundNearest to work properly. I can change it to RoundToZero.

Regarding the rounding modes, we can say it is similar to rounding of complex quotient like `round(a/b, r, r)`.

Reply:
You could also just accept `RoundNearest` by restricting `r` with `RoundingMode{:Nearest}`. I think it would make the most sense here though, to have separate rounding mode arguments for the real and imaginary part, to be consistent with `round`.